### ISSUE TITLE
Intel HD  wslg X Error:  unable to open display: 0

### ISSUE THREAD

**Issue by c13-github** [microsoft/wslg#375]
# Environment

 Win32NT             10.0.22000.0 Microsoft Windows NT 10.0.22000.0
Your Distribution version: 20.04
Your WSLg version: 1.0.24
GPU: Intel UHD 600
GPU Driver: Intel Corporation 27.20.100.820 (and earlier releases)

# Steps to reproduce
-Output of cat /mnt/wslg/versions.txt
WSLg ( x86_64 ): 1.0.24+Branch.main.Sha.b42023b5fed7567955f4c6c36123a5cf22e9292a
Mariner: VERSION="1.0.20210224"
FreeRDP: b05321cd4e6a862aef76163a69db4e1910245736
weston: ed4f3bbbf5efbd5bde880704bc03497f2446c3f2
pulseaudio: 2f0f0b8c3872780f15e275fc12899f4564f01bd5
mesa:

-Output of glxinfo
Error: unable to open display :0

**WSL logs**:  Manual log gathering crashes with wsl.wprp error:  Error code: 0xc00ce584
        DTD is prohibited.

        Line Number: 7
        Column Number: 11

Feedback hub doesn't appear to work for my build either

* Attach WSLg logs from  `/mnt/wslg`

You can access the wslg logs using explorer at: `\\wsl$\<Distro-Name>\mnt\wslg` (e.g.: `\\wsl$\Ubuntu-20.04\mnt\wslg`)

* `puseaudio.log`
* `weston.log`
* `versions.txt` : file rejected by GitHub as empty file
[pulseaudio.log](https://github.com/microsoft/wslg/files/6857788/pulseaudio.log)
[pulseaudio.log](https://github.com/microsoft/wslg/files/6857789/pulseaudio.log)
[weston.log](https://github.com/microsoft/wslg/files/6857790/weston.log)

#  Expected behavior
-Output confirming wslg is operating as expected

# Actual behavior
-Error: unable to open display:0
-OpenGL apps work correctly however

**Comment by c13-github** [comment]
[Xorg.0.log](https://github.com/microsoft/wslg/files/6857870/Xorg.0.log)
Also attaching Xorg.0 log

**Comment by onomatopellan** [comment]
Did you try with the new intel driver?
https://downloadcenter.intel.com/download/30579/Intel-Graphics-Windows-DCH-Drivers

**Comment by c13-github** [comment]
> Did you try with the new intel driver?
> https://downloadcenter.intel.com/download/30579/Intel-Graphics-Windows-DCH-Drivers

Yes correct. Windows Update then proceeded to suggest and install the current WHQL release version mentioned in my report.

In doing additional troubleshooting, I was able to install and then open Firefox by passing flag [MOZ_ENABLE_WAYLAND](https://discourse.ubuntu.com/t/using-firefox-wayland-on-mir-servers/19704), and this suggests to me that WSLg/other apps don't yet "see" Wayland.

**Comment by c13-github** [comment]
Feedback on this issue is that my experience with the Windows 11 dev channel build, WHQL Intel driver releases, Ubuntu-20.04 and WSLg version cited, GUI apps avoid Wayland for X.

Forcing a Weston install (e.g. `apt-get install weston`) and then forcing a Wayland preference:

- as mentioned for Firefox (with `export MOZ_ENABLE_WAYLAND=1`)
- for Chrome nightly stable with `-–enable-features=UseOzonePlatform –ozone-platform=wayland`

causes the expected "just works" WSLg GUI app behaviour. 

The glxinfo when any of above forced-mode apps are running remains: `glxinfo`
Error: unable to open display :0

If this is is not a WSLg bug, what may cause this behaviour, absent any random mucking about with WSLg moving parts, which wasn't intentional in my case at all?

**Comment by hideyukn88** [comment]
@c13-github, from your weston.log, it looks Xwayland (X server) is crashing, do you see any core dumps at /mnt/wslg/dumps? If Xwayland is crashing due to Intel driver, this could be dup of https://github.com/microsoft/wslg/issues/317. Btw, the latest Intel driver offered by WU doesn't support WSLg since Intel only release Beta driver of WSLg capable driver and it won't be offered via WU. Thanks for trying out WSLg!

>> [13:23:19.439] Spawned Xwayland server, pid 56
>> [13:23:20.211] xserver crashing too fast: 134

**Comment by c13-github** [comment]
> do you see any core dumps at /mnt/wslg/dumps? If Xwayland is crashing due to Intel driver, this could be dup of #317.

No dumps found. However I 'terminate' and 'shutdown' wsl every time. Does that clear them?

>Btw, the latest Intel driver offered by WU doesn't support WSLg since Intel only release Beta driver of WSLg capable driver and it won't be offered via WU.

This is interesting info. The beta Intel driver had the same behaviour,. WU did not mark WHQL ver. as an optional driver update, so I could not easily bypass installing it over top of the beta.

**Comment by hideyukn88** [comment]
Yes, once WSL VM is terminated by --shutdown or --terminate, the dumps are cleared.

**Comment by c13-github** [comment]
Thanks @hideyukn88  for assisting with Intel UHD crash reports I took a look at #317 and applied the debug `.wslconfig.` That didn't allow XWayland apps to open with just software GPU at all, and gave "unable to lock .. wayland-0.lock, maybe another compositor.. running" as error.

It's unclear if that is referring to a crashed instance of the compositor spawned with the app.

No dumps were generated again in my case, I checked before terminate and shutting down then renamed .wslconfig.

Currently, WSL+ Intel beta is in state where XWayland can't be forced open as before with flags. 

Now I must start app manually, leave the crashed session open and then start an app from another session. This suggests that WSLg in my case isn't Weston aware because as you indicated, can't successfully started and crashes right away. Yet it _is_ available to other logins.

The lines in the terminals of the screenshot I took will hopefully repro this odd behaviour fully.
![2_session_XWayland_app](https://user-images.githubusercontent.com/75766766/126818692-aadb2041-c003-487b-9364-a05f3fc83cd5.PNG)

**Comment by c13-github** [comment]
Closing the app (in this case Chrome) finally did leave behind an XWayland core dump. I will try to submit this using the Feedback tool if not I will post it here.

**Comment by hideyukn88** [comment]
@c13-github, first, regarding to ""unable to lock .. wayland-0.lock, maybe another compositor.. running", most likely this be possible because you manually installed weston in your distro. In WSLg model, the weston (which is specially customized for WSLg) is running is separate distro, called system-distro, please refer our WSLg architecture at README.md. Also WSLg does not use "weston-desktop-shell".

![image](https://user-images.githubusercontent.com/13857962/126822409-16aa4547-cb52-4787-a0f7-768e39ca4c01.png)

Would you please remove the manually installed weston (or any other X server solution if installed) and see if any dumps are created at /mnt/wslg/dumps ? thanks!

**Comment by c13-github** [comment]
Done and the behaviour persists now (2 sessions still needed to open an X-Wayland app), but no new core dumps. I submitted to Feedback Hub the previous core dump and used the bug subject as my feedback subject and referenced this issue. 

If this isn't the best way for that, could you advise best alternate means? 

I don't know if this is correct and helpful or not, but the logging timestamps in both windows seem to show that it is just one Weston instance handling all compositing, for both sessions.
![2_session_XWayland_app_weston_removed](https://user-images.githubusercontent.com/75766766/126826312-bdb26b8a-bc40-4dbb-8543-7aa7962cab9f.PNG)
